src/test/fixtures/patterns.json

```diff
@@ -0,0 +1,187 @@
+[
+    {
+       "name": "single segment pattern with leading slash and lone wildcard",
+       "pattern": "/*",
+       "paths": {
+          "foo": true,
+          "bar": true,
+          "foo/bar": false,
+          "foo/bar/baz": false
+       }
+    },
+    {
+       "name": "single segment pattern with leading slash and wildcard",
+       "pattern": "/f*",
+       "paths": {
+          "foo": true,
+          "foo/bar": true,
+          "foo/bar/baz": true,
+          "bar/foo": false,
+          "bar/foo/baz": false,
+          "bar/baz": false,
+          "xfoo": false
+       }
+    },
+    {
+       "name": "single segment pattern with trailing slash and wildcard",
+       "pattern": "f*/",
+       "paths": {
+          "foo": false,
+          "foo/bar": true,
+          "bar/foo": false,
+          "bar/foo/baz": true,
+          "bar/baz": false,
+          "xfoo": false
+       }
+    },
+    {
+       "name": "single segment pattern with leading and trailing slash and lone wildcard",
+       "pattern": "/*/",
+       "paths": {
+          "foo": false,
+          "foo/bar": true,
+          "bar/foo": true,
+          "bar/foo/baz": true
+       }
+    },
+    {
+       "name": "single segment pattern with leading and trailing slash and wildcard",
+       "pattern": "/f*/",
+       "paths": {
+          "foo": false,
+          "foo/bar": true,
+          "bar/foo": false,
+          "bar/foo/baz": false,
+          "bar/baz": false,
+          "xfoo": false
+       }
+    },
+    {
+       "name": "single segment pattern with escaped wildcard",
+       "pattern": "f\\*o",
+       "paths": {
+          "foo": false,
+          "f*o": true
+       }
+    },
+    {
+       "name": "pattern with trailing wildcard segment",
+       "pattern": "foo/*",
+       "paths": {
+          "foo": false,
+          "foo/bar": true,
+          "foo/bar/baz": false,
```

Comment:
This is the path that should not match according to the [codeowner documentation](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners):

```
# The `docs/*` pattern will match files like
# `docs/getting-started.md` but not further nested files like
# `docs/build-app/troubleshooting.md`.
docs/*  docs@example.com
```